calicoctl/resourcemgr/resourcemgr_test.go

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2019 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resourcemgr_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/projectcalico/calicoctl/calicoctl/resourcemgr"
+	api "github.com/projectcalico/libcalico-go/lib/apis/v3"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	DefaultIpPoolTemplate = `kind: IPPool
+apiVersion: projectcalico.org/v3
+metadata:
+  name: {{NAME}}
+spec:
+  cidr: {{CIDR}}
+  ipipMode: {{IPIPMODE}}
+  vxlanMode: {{VXLANMODE}}
+  natOutgoing: true
+`
+	MissingVxlanIpPoolTemplate = `kind: IPPool
+apiVersion: projectcalico.org/v3
+metadata:
+  name: {{NAME}}
+spec:
+  cidr: {{CIDR}}
+  ipipMode: {{IPIPMODE}}
+  natOutgoing: true
+`
+)
+
+var _ = Describe("Create resource from file", func() {
+	const CidrV6 = "2002::/64"
+	const CidrV4 = "192.168.0.0/16"
+	const PoolName = "my-ippool"
+	const AnotherPoolName = "another-ippool"
+
+	const VxlanModeNever = string(api.VXLANModeNever)
+	const VxlanModeAlways = string(api.VXLANModeAlways)
+	const IpipModeNever = string(api.IPIPModeNever)
+	const IpipModeAlways = string(api.IPIPModeAlways)
+
+	ipPoolV6WithNeverVxlan := ipPoolSpec(DefaultIpPoolTemplate, CidrV6, PoolName, VxlanModeNever, IpipModeNever)
```

Comment:
Are all these test supposed to test the generic feature of setting defaults and use ippool to prove that it works, or should the tests check all sort of combination of ipv4/6+vxlan+ipipmode ? Given the nature of the change, I would expect the former, but it looks like the latter.